libvirt/resource_libvirt_domain.go

```diff
@@ -599,7 +599,8 @@ func resourceLibvirtDomainCreate(d *schema.ResourceData, meta interface{}) error
 			for _, addressI := range addressesI.([]interface{}) {
 				address := addressI.(string)
 				log.Printf("[INFO] Finally adding IP/MAC/host=%s/%s/%s", address, mac, pending.hostname)
-				updateOrAddHost(pending.network, address, mac, pending.hostname)
+				err := updateOrAddHost(pending.network, address, mac, pending.hostname)
+				pending.network.Free()
 				if err != nil {
 					return fmt.Errorf("Could not add IP/MAC/host=%s/%s/%s: %s", address, mac, pending.hostname, err)
 				}
```

Comment:
Good catch, `err` wasn't simply associated before :man_facepalming: 

I would refactor the code in this way:

```golang
if err := updateOrAddHost(pending.network, address, mac, pending.hostname); err != nil {
   // log the message somehow, do not return with an error - don' return the `error` object
}
if err := network.Free(); err != nil {
  // log error message, move forward
}
```

I would not return in the middle of this for loop because that would prevent to process other pending addresses if one of them fails for whatever reason. Exiting from the loop would also prevent the de-allocation of the objects.